doc/BMI_MODELS.md

```diff
@@ -185,6 +201,64 @@ This is needed both due to the design of the **C** language variant of BMI, and
 
 Future versions of NextGen will provide alternative ways to declaratively configure function names from a BMI C library so they can individually be dynamically loaded.
 
+## BMI Models Written in C++
+
+- [BMI C++ Model As Shared Library](#bmi-c-model-as-shared-library-1)
+  - [Dynamic Loading](#dynamic-loading-1)
+  - [Additional Bootstrapping Functions Needed](#additional-bootstrapping-functions-needed)
+    - [Why?](#why-1)
+- [BMI C++ Example](#bmi-c-example)
+
+You can implement a model in C++ by writing an object which implements the [BMI C++ interface](https://github.com/csdms/bmi-cxx).
+
+### BMI C++ Model As Shared Library
+
+For **C++** models, the model should be packaged as a pre-compiled shared library. Support for loading of C++ modules/libraries is always enabled, so no build system flags are required. 
+#### Dynamic Loading
+
+As noted [above](#semi-optional-parameters), the path to the shared library must be provided in the configuration so that the module can be loaded at runtime.
+
+#### Additional Bootstrapping Functions Needed
+
+BMI models written in **C++** should implement two **C** functions declared with `extern "C"`. These functions instantiate and destroy a **C++** BMI model object. By default, these functions are expected to be named `bmi_model_create` and `bmi_model_destroy`, and have signatures like the following:
+
+    extern "C"
+    {
+      /**
+      * @brief Construct this BMI instance as a normal C++ object, to be returned to the framework.
+      * @return A pointer to the newly allocated instance.
+      */
+      MyBmiModelClass *bmi_model_create()
```

Comment:
@mattw-nws, @hellkite500, wouldn't it be more appropriate for the return type (and param type for the destroyer) to be `::bmi::Bmi`?

Reply:
Hmm.  As long as it is fully virtual, then I think that will work, otherwise you risk pointer slicing.